src/app/modules/color-scheme/color-bar/color-bar.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
+import { ColorScheme } from '../color-scheme-popup/color-scheme-popup.component';
+
+@Component({
+  selector: 'ccf-color-bar',
+  templateUrl: './color-bar.component.html',
+  styleUrls: ['./color-bar.component.scss']
+})
+export class ColorBarComponent implements OnInit {
+
+  @Input() schemeOptions: ColorScheme[];
+  @Input() colorScheme: ColorScheme;
+  @Input() schemeIndex: number;
+  @Output() colorChange = new EventEmitter<number[]>();
+
+  gradientstyle: string;
+  colorSelectedStatus: boolean[][] = [];
+  colorIndex: number[];
+
+  ngOnInit() {
+    this.resetColorStatus();
+    this.colorIndex = this.colorScheme.type === 'discrete' ? [0,1,2,3,4,5,6] : [0];
+    this.gradientstyle = this.colorScheme.type === 'gradient' ? `linear-gradient(to right, ${this.getColor(0)} , ${this.getColor(1)}, ${this.getColor(2)})` : 'none';
```

Comment:
Linear gradients may have more or less that 3 colors. Also it needs to use the percentages in the scheme to determine the changing points of the colors. Check demo example 3 for percentage usage: https://developer.mozilla.org/en-US/docs/Web/CSS/linear-gradient